library/ssl_cache.c

```diff
@@ -43,6 +43,11 @@
 
 #include <string.h>
 
+/* Implementation that should never be optimized out by the compiler */
+static void mbedtls_zeroize( void *v, size_t n ) {
```

Comment:
I don't think that this is necessary, because if you use `cache->chain` later, then the compiler can't optimise it out and it really doesn't matter otherwise.